### DISCUSSION TITLE
Datetime Difference

### DISCUSSION THREAD

**Discussion by ikaheraJQ** [Budibase/budibase#5388] [Help] [answered]
Sorry, I want to ask you why did I get 2022-04-11T17:00:00.000Z instead of 2022-04-12T17:00:00.000Z from Datepicker when I select April 12, 2022? I need to filter data using date range, so it will be a problem if the date received by the system is not exactly the same value with the date I have choosen.
Thank you for your help 🙏🏽

<img width="664" alt="Screen Shot 2022-04-13 at 14 59 51" src="https://user-images.githubusercontent.com/82486031/163130629-097405a1-529b-4c9e-ac48-57570ce423c5.png">

**Comment by melohagan** [comment]
Hi @ikaheraJQ 

Would you mind showing the binding you have on your disabled field please?

**Reply by ikaheraJQ** [reply]
Here

<img width="1159" alt="Screen Shot 2022-04-13 at 16 09 06" src="https://user-images.githubusercontent.com/82486031/163142015-d7660034-73be-420f-be66-496ecd2672f1.png">

**Comment by melohagan** [comment]
@ikaheraJQ Thanks for your post. It may be an issue with timezones. Out of curiosity what is your timezone? 

I have added a comment to this issue which is exploring date picker improvements: https://github.com/Budibase/budibase/issues/5102#issuecomment-1098081268

**Reply by ikaheraJQ** [reply]
My timezone is UTC + 7

**Comment by aptkingston** [comment]
When you pick a date only (without a time component) it will pick the time 00:00. You must be in a timezone that is UTC+9, as the timestamp it sent up is the correct UTC timestamp for 12th April 00:00 in your timzone. The `Z` at the end denotes that it is a UTC timestamp. If you are sending the value to any backend that understands timezones (like most SQL table date fields) then everything will be fine and this is correct. If you are sending this to a backend that does not understand timezones then this will be a problem.

We have discussed this lots and it is my opinion that we need a plain date data type that does not do any timezone conversions.

**Comment by ikaheraJQ** [comment, marked as answer]
Okay, thanks for your help, @melohagan and @aptkingston 🙏🏽
I have solved this problem by changing the binding to {{ date Form.Fields.searchByRegDateFrom "YYYY/MM/DD" }}

**Reply by melohagan** [reply]
Awesome, good thinking! :)
